usermods/sht/usermod_sht.h

```diff
@@ -0,0 +1,312 @@
+#pragma once
+
+#include "SHT85.h"
+
+
+class ShtUsermod : public Usermod
+{
+  private:
+    bool enabled = false;
+    bool firstRunDone = false;
+    bool initDone = false;
+    bool haMqttDiscovery = false;
+    SHT *shtTempHumidSensor;
+
+    // SHT vars
+    bool shtInitDone = false;
+    bool shtReadDataSuccess = false;
+    byte shtI2cAddress = 0x44;
+    unsigned long shtLastTimeUpdated = 0;
+    bool shtDataRequested = false;
+    float shtCurrentTemp = 0;
+    float shtLastKnownTemp = 0;
+    float shtCurrentHumidity = 0;
+    float shtLastKnownHumidity = 0;
+
+
+    void initShtTempHumiditySensor()
+    {
+      PinManagerPinType pins[2] = { { i2c_sda, true }, { i2c_scl, true } };
+      if (!pinManager.allocateMultiplePins(pins, 2, PinOwner::HW_I2C)) {
+        DEBUG_PRINTF("[%s] SHT pin allocation failed!\n", _name);
+        shtInitDone = false;
+        cleanupShtTempHumiditySensor();
+        cleanup();
+        return;
+      }
+
+      #ifdef USERMOD_SHT_TYPE_SHT31
```

Comment:
Can you make this runtime configurable?
It will require de-init on parameter change but I can see you already implemented de-init.
Example of usermod config drop-down list can be seen in 4LD ALT (`appendConfigData()`).
You may as well query I2C to detect sensor type if this is available.